File: Lesson4-counting_elements/MaxCounters/mysol.c
// you can write to stdout for debugging purposes, e.g.
// printf("this is a debug message\n");
#include <string.h>
#define set_max(c, n, max) for(int j =0; j<n; j++) c[j]=max

struct Results solution(int N, int A[], int M) {
    struct Results result;
    // write your code in C99 (gcc 6.2.0)
    int i = 0, max = 0;

    result.C = malloc(sizeof(int)*N);
    memset(result.C, 0, sizeof(int)*N);

    for (i=0; i<M; i++){
        if (A[i]>=N)
            set_max(result.C, N, max);
        else{
            result.C[A[i]-1]++;
            if (result.C[A[i]-1] > max)
                max = result.C[A[i]-1];
        }
        // printf("C:[");
        // for(int k=0; k<N; k++) 
        //     printf(" %d", result.C[k]);
        // printf("] \n");
    }
    result.L = N;
    return result;
}

File: Lesson6-sorting/MaxProductOfThree/mySol.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    if len(A)==3:
        return A[0]*A[1]*A[2]
    A.sort(reverse=True)

    ret1 = A[0]*A[1]*A[2]
    ret2 = A[0]*A[-1]*A[-2]
    
    return ret1 if ret1>=ret2 else ret2